llvm/test/tools/sycl-post-link/sycl-external-funcs/split-per-source.ll

```diff
@@ -0,0 +1,28 @@
+; RUN: sycl-post-link -split=source -symbols -S %s -o %t.table
+; RUN: FileCheck %s -input-file=%t_0.ll --check-prefixes CHECK-IR1
+; RUN: FileCheck %s -input-file=%t_1.ll --check-prefixes CHECK-IR2
+; RUN: FileCheck %s -input-file=%t_0.sym --check-prefixes CHECK-SYM1
+; RUN: FileCheck %s -input-file=%t_1.sym --check-prefixes CHECK-SYM2
+
+; This test checks that unreferenced functions are not dropped
+; from the module after splitting in per-source mode.
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64-unknown-linux-sycldevice"
+
+define dso_local spir_func void @externalDeviceFunc() #0 {
```

Comment:
Can we add a test where some SYCL_EXTERNAL function `foo` is defined in source "a.cpp" but it is also referenced by a kernel defined in source "b.cpp"? In this case design of dynamic code linking says that definition of function `foo` should be present in both resulting modules when per-source split is requested.

Reply:
Added.